#VRML_SIM R2023b utf8
# Describe the functionality of your PROTO here.


PROTO BraitenbergBot [
  field SFVec3f translation 0 0 0
  field SFRotation rotation 0 0 1 0
  field SFString name "robot"
  field SFString controller "<generic>"
]
{
  Robot {
    translation IS translation
    rotation IS rotation
    name IS name
    controller IS controller
    children [
      Transform {
        translation 0 0 0.0415
        rotation 0 0 1 -1.570796
        children [
          Shape {
            appearance PBRAppearance {
              baseColor 0 0.4 0.8
              roughness 1
              metalness 0
            }
            geometry DEF BODY Cylinder {
              height 0.08
              radius 0.045
            }
          }
        ]
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          position 18.1118551202026
          axis 0 1 0
          anchor 0 0.045 0.025
        }
        device [
          RotationalMotor {
            name "left_wheel_motor"
          }
          PositionSensor {
            name "left_wheel_sensor"
          }
        ]
        endPoint Solid {
          translation 0 0.045 0.025
          rotation -0.4355084909218354 0.435508576269092 0.7878227175774091 4.476136003703759
          children [
            DEF WHEEL Transform {
              rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
              children [
                Shape {
                  appearance PBRAppearance {
                    baseColor 1 0 0
                    roughness 1
                    metalness 0
                  }
                  geometry Cylinder {
                    height 0.01
                    radius 0.025
                  }
                }
              ]
            }
          ]
          name "left wheel"
          boundingObject USE WHEEL
          physics DEF PHYSICS_WHEEL Physics {
            density -1
            mass 0.05
          }
          linearVelocity -5.259033105332559e-07 -3.959179196243546e-07 -1.9005429407780097e-09
          angularVelocity -5.978128712640519e-06 -1.926862268122827e-05 -1.1953703549325266e-05
        }
      }
      HingeJoint {
        jointParameters HingeJointParameters {
          position -18.111678457347093
          axis 0 1 0
          anchor 0 -0.045 0.025
        }
        device [
          RotationalMotor {
            name "right_wheel_motor"
          }
          PositionSensor {
            name "right_wheel_sensor"
          }
        ]
        endPoint Solid {
          translation 0 -0.045 0.025
          rotation 0.385817011261429 -0.3858169816805515 0.8380277384837611 4.536597276216522
          children [
            USE WHEEL
          ]
          name "right wheel"
          boundingObject USE WHEEL
          physics USE PHYSICS_WHEEL
          linearVelocity 5.290480022231787e-07 1.6738477831470564e-07 -3.0536537423167124e-08
          angularVelocity 6.37787690013977e-06 2.297477673672875e-05 -1.1953675604291352e-05
        }
      }
      DistanceSensor {
        translation 0.042 0.02 0.063
        rotation 0 0 -1 -0.5
        children [
          DEF SENSOR Transform {
            rotation 0 1 0 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 1 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.004
                  radius 0.008
                }
              }
            ]
          }
        ]
        name "ds0"
        lookupTable [
          0    1024 0
          0.05 1024 0
          0.15 0    0
        ]
        numberOfRays 2
        aperture 1
      }
      LightSensor {
        translation 0.042 0.02 0.063
        rotation 0 0 -1 -0.5
        children [
          DEF LIGHT_SENSOR Transform {
            rotation 0 1 0 1.57
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 1
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.005
                  radius 0.002
                }
              }
            ]
          }
        ]
        name "ls0"
        lookupTable [
          0 0    0
          3 1000 0
        ]
        occlusion TRUE
      }
      DistanceSensor {
        translation 0.042 -0.02 0.063
        rotation 0 0 1 -0.5
        children [
          USE SENSOR
        ]
        name "ds1"
        lookupTable [
          0    1024 0
          0.05 1024 0
          0.15 0    0
        ]
        numberOfRays 2
        aperture 1
      }
      LightSensor {
        translation 0.042 -0.02 0.063
        rotation 0 0 1 -0.5
        children [
          USE LIGHT_SENSOR
        ]
        name "ls1"
        lookupTable [
          0 0    0
          3 1000 0
        ]
        occlusion TRUE
      }
    ]
    boundingObject Transform {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        USE BODY
      ]
    }
    physics Physics {
      density -1
      mass 0.5
    }
  }
}
